src/fe/fe_base.C

```diff
@@ -742,7 +742,65 @@ void FEGenericBase<Real>::compute_dual_shape_coeffs ()
 
   //compute dual basis coefficient (dual_coeff)
   dual_coeff.resize(sz, sz);
+  dual_coeff.zero();
+
+  if (fefamily == LAGRANGE && Dim == 1)
+  {
+    // for 1D LAGRANGE, we have pre-computed accurate dual shape coefficients
+    libmesh_error_msg_if(sz > 4, "ERROR:  Unsupported shape function size " << sz);
+    if (sz == 2)
+    {
+      dual_coeff(0,0) = 2.;
+      dual_coeff(0,1) = -1.;
+      dual_coeff(1,0) = -1.;
+      dual_coeff(1,1) = 2.;
+    }
+    if (sz == 3)
+    {
+      dual_coeff(0,0) = 1.5;
+      dual_coeff(0,1) = 0.5;
+      dual_coeff(0,2) = -0.25;
+      dual_coeff(1,0) = 0.5;
+      dual_coeff(1,1) = 1.5;
+      dual_coeff(1,2) = -0.25;
+      dual_coeff(2,0) = -1.;
+      dual_coeff(2,1) = -1.;
+      dual_coeff(2,2) = 1.5;
+    }
+    if (sz == 4)
+    {
+      dual_coeff(0,0) = 2.;
+      dual_coeff(0,1) = -0.5;
+      dual_coeff(0,2) = -0.314814814814815;
+      dual_coeff(0,3) = 0.148148148148148;
+      dual_coeff(1,0) = -0.5;
+      dual_coeff(1,1) = 2.;
+      dual_coeff(1,2) = 0.148148148148148;
+      dual_coeff(1,3) = -0.314814814814815;
+      dual_coeff(2,0) = -0.944444444444444;
+      dual_coeff(2,1) = 0.444444444444443;
+      dual_coeff(2,2) = 1.14403292181070;
+      dual_coeff(2,3) = 0.0226337448559681;
+      dual_coeff(3,0) = 0.444444444444444;
+      dual_coeff(3,1) = -0.944444444444444;
+      dual_coeff(3,2) = 0.0226337448559681;
+      dual_coeff(3,3) = 1.14403292181070;
+    }
+  }
+  else
+    // for the other cases, we rely on the numerical integration of the primal basis
+    // to obtain the dual shape coefficients
+    compute_numerical_dual_shape_coeffs();
+}
+
+template <>
+void FEGenericBase<Real>::compute_numerical_dual_shape_coeffs ()
+{
+  unsigned int sz=phi.size();
+  libmesh_error_msg_if(!sz, "ERROR:  dual basis should be computed after the primal basis");
+
   DenseMatrix<Real> A(sz, sz), D(sz, sz);
+  A.zero(); D.zero();
 
   const std::vector<Real> JxW = this->get_JxW();
```

Comment:
This line is not technically part of your diff, but I think it should probably be catching a *reference*, i.e.
```
const std::vector<Real> & JxW = this->get_JxW();
```
and not making a copy as it is doing currently...